src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -969,6 +971,22 @@ def aks_upgrade(cmd,    # pylint: disable=unused-argument, too-many-return-state
         mc = client.get(resource_group_name, name)
         return _remove_nulls([mc])[0]
 
+    if cluster_snapshot_id:
+        CreationData = cmd.get_models(
+            "CreationData",
+            resource_type=CUSTOM_MGMT_AKS_PREVIEW,
+            operation_group="managed_clusters",
+        )
+        instance.creation_data = CreationData(
+            source_resource_id=cluster_snapshot_id
+        )
+        mcsnapshot = get_cluster_snapshot_by_snapshot_id(cmd.cli_ctx, cluster_snapshot_id)
+        if not mcsnapshot:
+            raise CLIError(
+                "Given managed cluster snapshot not exsit, input id: {}".format(cluster_snapshot_id)
+            )
+        kubernetes_version = mcsnapshot.managed_cluster_properties_read_only.kubernetes_version
```

Comment:
If the version in the mcsnapshot is lower than the current cluster, will this be acceptable?

Reply:
No, it will be blocked at RP frontend validation.